core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/etcs/ETCSBrakingCurves.kt

```diff
@@ -368,10 +379,11 @@ private fun keepBrakingCurveUnderOverlay(
             EnvelopeConstraint(overlay, EnvelopePartConstraintType.CEILING)
         )
     overlayBuilder.initEnvelopePart(positions[nbPoints - 1], speeds[nbPoints - 1], -1.0)
-    for (i in fullBrakingCurve.pointCount() - 2 downTo 0) {
+    for (i in nbPoints - 2 downTo 0) {
         if (!overlayBuilder.addStep(positions[i], speeds[i], timeDeltas[i])) break
     }
-    if (partBuilder.stepCount() <= 1) return null
+    if (partBuilder.stepCount() <= 1)
```

Comment:
`stepCount()` is based on `times`, and there is 1 less value than positions:
```suggestion
    if (partBuilder.isEmpty)
```
So this failed when there was a stop close from the start, and a curve with only 2 positions (19 first meters in my test).

Reply:
Nice catch!!